homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -82,11 +89,39 @@ async def async_setup_entry(
             continue
         room_id = shade.raw_data.get(ROOM_ID_IN_SHADE)
         room_name = room_data.get(room_id, {}).get(ROOM_NAME_UNICODE, "")
-        entities.append(
-            PowerViewShade(
-                coordinator, device_info, room_name, shade, name_before_refresh
+
+        if shade.shade_type.shade_type in TOPDOWN_SHADES:
```

Comment:
Make a list of shade types constants and loop through it to build the list of entities.  Then you can do a loop with this in the inner portion

```
            entities.append(
                PowerViewShade(
                    coordinator,
                    device_info,
                    room_name,
                    shade,
                    name_before_refresh,
                    shade_type,
            )
```

Reply:
Coming back to this after reviewing below, you could also have a different class for top and bottom `PowerViewTopBottomShade` and `PowerViewStandardShade`

Then make `PowerViewShade` the base class so you can split the code without having to do lots of branching

Reply:
Just kept the vase shade as PowerViewShade an Extended Class for TDBU Shades